### ISSUE TITLE
Graph format: $node variable is not initialized in the else block

### ISSUE THREAD

**Issue by jewloszewicz** [SemanticMediaWiki/SemanticResultFormats#549]
### Setup

- MW version: 1.31.5
- SMW version: 3.1.0
- SRF version: 3.2.0-alpha

### Issue

Error from line 186 of /var/www/smw/extensions/SemanticResultFormats/src/Graph/GraphPrinter.php: Call to a member function addParentNode() on null

**Comment by JeroenDeDauw** [comment]
Thanks for reporting this issue!

Ping @SbstnS

**Comment by SbstnS** [comment]
@jewloszewicz can you please post your inline query which causes this error so i can reproduce it.

**Comment by SbstnS** [comment]
@jewloszewicz I'm working with the same setup as you've described above but I never run into this error. 

You may have made some mistakes during installing the extension.
Please check the installation guide.

https://www.semantic-mediawiki.org/wiki/Extension:Semantic_Result_Formats/Installation

**Comment by jewloszewicz** [comment]
@SbstnS Unfortunately, I don't have access to that instance anymore.

There is mistake in the code. $node in the else statement is never initialized.

https://github.com/SemanticMediaWiki/SemanticResultFormats/blob/1b983766322732e50fa6f613e006c223f7bd9533/src/Graph/GraphPrinter.php#L183

**Comment by SbstnS** [comment]
@jewloszewicz I'm not sure how then you can run into this error that you've described above....

`Error from line 186 of /var/www/smw/extensions/SemanticResultFormats/src/Graph/GraphPrinter.php: Call to a member function addParentNode() on null`

To me everything works as expected! 
If i had your inline query than i could cover the hole test scenario...

Here is the one that I used for testing.

```
{{#ask:[[Category:Team]]
  |?Part Of Team
  |format=graph
  |graphcolor=yes
  |graphlink=yes
  |graphname=PeopleKnows
  |graphlegend=yes
  |graphlabel=yes
  |graphsize=10,100
  |nodelabel=displaytitle
}}
```

**Comment by SbstnS** [comment]
@JeroenDeDauw 
But I noticed a few parts while testing the code that could be improved. I'll do this in a new PullRequest.

**Comment by alex-mashin** [comment]
I think, this bug has benn fixed by  #749.
